RocketChat/Rocket.Chat
# fix: Attachments not collapsing when using incoming webhook

## Proposed changes (including videos or screenshots)


Right now there has been a long-standing issue where attachments don't collapse even when `"collapsed": true` is specified. Collapse always initializes as null (doesn't consider the value from the incoming webhook).

Before: (regardless of specifying collapsed: true or collapsed: false)
![image](https://github.com/RocketChat/Rocket.Chat/assets/98163873/f3d0ede6-9e25-4372-9800-6ce13e8fee40)


After:

https://github.com/RocketChat/Rocket.Chat/assets/98163873/a2fcc6ad-853c-48fe-970a-291885d2af0d





## Issue(s)


fix #29276 

## Steps to test or reproduce


1. Create webhook
2. Try to send a message containing an attachment
3. Specify "collapsed": true
4. The attachment should be collapsed but it's not.
5. Specify "collapsed": false
6. The attachment is not collapsed.

## Further comments


`curl -H "X-Auth-Token: my-token" \ -H "X-User-Id: my-id" \ -H "Content-type:application/json" \ https://localhost:3000/api/v1/chat.postMessage \ -d '{"channel":"#general","emoji":":smirk:","text":"Sample message","attachments":[{"collapsed":true,"color":"#abcdef","fields":[{"title":"test 1","value":"test 1 value"},{"title":"test 2","value":"test 2 value"}],"text":"test text","title":"test title","title_link":"https://youtube.com","ts":"2016-12-09T16:53:06.761Z"}]}'` 

This command can be used to test the modification
